### ISSUE TITLE
Overrides keys missmatch

### ISSUE THREAD

**Issue by ThomasBrierley** [ksmithut/bower-files#71]
Bower "name" properties appear to be used to match "overrides" keys, where as registry names are used to match "dependencies" keys. There is no requirement for the "name" property to match the bower registry name, just like the infamous, redundant and depreciated "version" property.

This caught me out when trying to add some missing font overrides for katex. The registry name is "katex" and the bower.json name is "KaTeX", impossible to know without digging into the dependency's bower.json:

Silently Broken:
``` JSON
{
	"dependencies": {
		"katex": "^0",
	},
	"overrides": {
		"katex": {
			"main": "foo.js"
		}
	}
}
```

Working:
``` JSON
{
	"dependencies": {
		"katex": "^0",
	},
	"overrides": {
		"KaTeX": {
			"main": "foo.js"
		}
	}
}
```

Is the amalgamation of main files dealt with by **bower-files** or **bower** ? Either way it's going to be less breaky if overrides keys can be derived from the registry, if not possible the directory names could be used.

**Comment by ksmithut** [comment]
Ah, good catch. I'll look into this and fix this. I'll keep you updated.

**Comment by ksmithut** [comment]
hmmm... interesting. When I run `bower install --save katex`, I get the following in my bower.json:

```js
"dependencies": {
  "KaTeX": "katex#^0.6.0"
}
```

I can confirm that using yours causes the specified behavior. Looks like they need to be more bower spec compliant as well:

```
bower invalid-meta  The "name" is recommended to be lowercase, can contain digits, dots, dashes
bower invalid-meta  The "main" field cannot contain minified files
bower invalid-meta  The "main" field cannot contain minified files
```

I'm still going to attempt to fix this because it's frustrating that it fails silently.

**Comment by ThomasBrierley** [comment]
Ah, I usually manually enter them in because I want to specify the semver upgradability bit. Interesting how the key is the bower.json name and the value is the registry name (which I usually omit).

You are correct katex should fix their name, but to be fair this part of the spec is broken, there just shouldn't be a name property at all so that there is only one definitive way to refer to a repo.

I'm wondering if the overrides should try to match against both so that the fix is a non breaking change.

**Comment by ksmithut** [comment]
yup, that's what I'm going to do. It will pull in the overrides for both the name, and the directory name.

**Comment by ksmithut** [comment]
Alright, I've published 3.14.1 with this fix. Thanks for reporting!

**Comment by ThomasBrierley** [comment]
Works great, Thanks @ksmithut
